homeassistant/components/sensor/nissan_leaf.py

```diff
@@ -0,0 +1,116 @@
+"""Battery Charge and Range Support for the Nissan Leaf.
+
+Please refer to the main platform component for configuration details
+"""
+
+import logging
+from homeassistant.util.unit_system import IMPERIAL_SYSTEM, METRIC_SYSTEM
+from homeassistant.util.distance import LENGTH_KILOMETERS, LENGTH_MILES
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.components.nissan_leaf import (
+    DATA_BATTERY, DATA_CHARGING, DATA_LEAF, DATA_RANGE_AC, DATA_RANGE_AC_OFF,
+    LeafEntity
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['nissan_leaf']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sensors setup."""
+    _LOGGER.debug("setup_platform nissan_leaf sensors, discovery_info=%s",
+                  discovery_info)
+
+    devices = []
+    for key, value in hass.data[DATA_LEAF].items():
+        _LOGGER.debug("adding sensor for item key=%s, value=%s", key, value)
+        devices.append(LeafBatterySensor(value))
+        devices.append(LeafRangeSensor(value, True))
+        devices.append(LeafRangeSensor(value, False))
+
+    add_devices(devices, True)
+
+
+class LeafBatterySensor(LeafEntity):
+    """Nissan Leaf Battery Sensor."""
+
+    @property
+    def name(self):
+        """Sensor Name."""
+        return self.car.leaf.nickname + " Charge"
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return 'battery'
```

Comment:
use DEVICE_CLASS_BATTERY from const

Reply:
Thanks. Fixed in commit https://github.com/home-assistant/home-assistant/pull/19786/commits/72924b812409f026ae6a7ec694f44c0cd9e1c14a